tests/unit/test_charm.py

```diff
@@ -876,3 +876,43 @@ def test_wordpress_version_set(harness: ops.testing.Harness):
     harness.begin_with_initial_hooks()
 
     assert harness.get_workload_version() == WordpressContainerMock._WORDPRESS_VERSION
+
+
+@pytest.mark.usefixtures("attach_storage")
+def test_waiting_for_leader_installation_timeout(
+    patch: WordpressPatch, harness: ops.testing.Harness, app_name
+):
+    """
+    arrange: charm peer and database relation is ready, the storage is attached.
+    act: start the charm as a follower unit.
+    assert: charm unit should enter blocked state, and the installation error should be seen
+        in the status.
+    """
+    replica_relation_id = harness.add_relation("wordpress-replica", app_name)
+    harness.update_relation_data(
+        relation_id=replica_relation_id,
+        app_or_unit=app_name,
+        key_values={k: "test" for k in WordpressCharm._wordpress_secret_key_fields()},
+    )
+    db_relation_id = harness.add_relation("database", "mysql")
+    harness.add_relation_unit(db_relation_id, "mysql/0")
+    harness.update_relation_data(
+        relation_id=db_relation_id,
+        app_or_unit="mysql",
+        key_values={
+            "endpoints": "test",
+            "database": "test",
+            "username": "test",
+            "password": "test",
+        },
+    )
+    # ignore bandit B106:hardcoded_password_funcarg
+    patch.database.prepare_database(
+        host="test", database="test", user="test", password="test"
```

Comment:
Why not use the `secrets` library to generate a random string for the password? That way we don't need a disable for a security issue

Reply:
I think the problem of using `secrets` library is that it's a little bit perplexing for people not knowing the intent. I once received a comment asking about why using `secrets.token_urlsafe` in tests.

Reply:
The explanation can be that it is a good habit not to hardcode passwords. Whilst the harm done in tests is not that bad, people could still copy paste code from tests or see it and think it is reasonable. It is also good to keep the security warning disables to a minimum so we only do it rarely and treat warnings seriously. You can always include a comment explaining why we use secrets to generate the password

Reply:
Gotcha, updated, thanks!